In [44]:
import pandas as pd
import numpy as np
from scipy import stats

In [45]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [46]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

## 1. What test is appropriate for this problem? Does CLT apply?

#### Central Limit Theorem applies because even though the population distributions are categorical, the sampling distribution of proportions will be approximately normal. We would have to use the Chi-Square Test for Independence. This test is applied when you have two categorical variables from a single population and is used to determine whether there is a significant association between the two variables.

## 2. What are the null and alternate hypotheses?

 #### The Null Hpothesis generated from this data is as follows:
 #### $H_{0}:$ black or white sounding names have no impact on callbacks from employers
 
 #### $H_{A}:$ Race does have a significant impact on callbacks from employers
 

In [47]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)
sum(data[data.race=='w'].call)

black_callbacks = data[data.race=='b'].call
white_callbacks = data[data.race=='w'].call

In [48]:
print("black sounding name callback rate: ", sum(black_callbacks[black_callbacks == 1])/len(black_callbacks))
print("white sounding name callback rate: ", sum(white_callbacks[white_callbacks == 1])/len(white_callbacks))

black sounding name callback rate:  0.064476386037
white sounding name callback rate:  0.0965092402464


#### Approximately 9.7% of  white sounding names got callbacks from employers while 6.4 black sounding names got callbacks. We have to now determine whether this disparity is statistically significant.

## 3. Compute margin of error, confidence interval, and p-value.

In [49]:
table = pd.crosstab(data.race, data.call)
table.columns = ['No','Yes']
table.index = ['Black','White']
table

No  Yes
Black  2278  157
White  2200  235

#### $X^{2}$ and P-Value:

In [50]:
chi2, p_value, data, ex = stats.chi2_contingency(cont_table)

print("Test Stat:", chi2)
print("p-value:",p_value)

Test Stat: 16.4490285842
p-value: 4.99757838996e-05


#### The Chi Squared Test Statistic and the P-Value show that race and callbacks are dependent. Chi Squared distributions are bounded at 0 and have a tail that approaches the x-axis the larger the x value is. Tests using a Chi Squared distribution use a one-way test which require higher values to prove statistical significance. As a result, they rely on a critical value, instead of a broader confidence interval. The critical value encompasses the margin of error and confidence interval:

In [51]:
chi2_critical = stats.chi2.ppf(q=0.95, df=data)

print("Chi-square Critical value:", chi2_critical)
print("The Chi Squared Test Stat: ", chi2, "| Critical Value: ", chi2_critical)

Chi-square Critical value: 3.84145882069
The Chi Squared Test Stat:  16.4490285842 | Critical Value:  3.84145882069


#### Since the Chi Squared Test Stat is larger than the Critical Value, this proves the Alternative Hypothesis to be true: there is evidence that shows callbacks are dependent on race.

## 4. Write a story describing the statistical significance in the context or the original problem.

## Is Race a Factor of Employer CallBacks?

#### Of the provided test data, we found that the proportion of callbacks white sounding names getting callbacks from employers compared to black sounding names were 9.7% to 6.4% (a disparity of 3.3%). When we performed a Chi Squared Test to see if this difference was statistically significant, the resulting test statistic of 16.4 was larger than the critical value of 3.8, meaning that the difference was in fact statistically significant and that race did play a factor in employee callbacks. 

#### Just how significant the impact was could not be measured and intersectionality does play a role in callbacks as well. Since we only isolated for the race and callback variables in the provided data, it is not certain just how significant the impact of race is callbacks.

## 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

#### Since we have isolated for only the race and callback portions of the dataset, we cannot fully conclude that race is the most important factor in callback success. Intersectionality between economic background, education, qualification, family, and a multitide of other factors can be present that determine job prospects for individuals. The analysis must qualify the statements' absolutes and wording should be wary of attributing one factor as the most important indicator of callback failure/success.